src/Microsoft.SqlTools.ServiceLayer/QueryExecution/Query.cs

```diff
@@ -89,7 +122,32 @@ public Query(string queryText, ConnectionInfo connection, QueryExecutionSettings
                             batch.EndLocation.LineNumber - 1,
                             batch.EndLocation.ColumnNumber - 1),
                         index, outputFactory));
+                        
             Batches = batchSelection.ToArray();
+
+            // Create our batch lists
+            BeforeBatches = new List<Batch>();
+            AfterBatches = new List<Batch>();
+
+            // Checking settings for execution plan options 
+            if (querySettings.ExecutionPlanOptions.IncludeEstimatedExecutionPlanXml) 
+            {
+                // Check support level 
+                if (this.doesSupportExecutionPlan(connection))
+                {
+                    // Enable set showplan xml
+                    addBatch(string.Format(s_SetShowPlanXml, s_On), BeforeBatches, streamOutputFactory);
```

Comment:
>s_On [](start = 61, length = 4)

these are duplicate code, move the login to a method or even another class to handle options